### PR TITLE
fix: update osmosis min fee to comply with v15.0.0 min consensus fee update

### PR DESCRIPTION
**Pull request by kaladinlight**
- single const
- use `MIN_FEE` in osmosis swapper
- use `MIN_FEE` with `calcFee` + `scalar` for `getFeeData`

### CODE REVIEW

**Review by pastaghost — DISMISSED** [review]
LGTM!

File: packages/chain-adapters/src/cosmossdk/CosmosSdkBaseAdapter.ts [hunk 1]
```diff
@@ -57,6 +58,16 @@ export const assertIsValidatorAddress = (validator: string, chainId: CosmosSdkCh
   }
 }
 
+type ConfirmationSpeed = 'slow' | 'average' | 'fast'
```
Comment (gomesalexandre) [inline packages/chain-adapters/src/cosmossdk/CosmosSdkBaseAdapter.ts:61]: Could reuse `FeeDataKey` here?

**Review by gomesalexandre — APPROVED** [review]
Untested but conceptually LGTM

File: packages/chain-adapters/src/cosmossdk/CosmosSdkBaseAdapter.ts (hunk 1 above)
Comment (kaladinlight) [reply packages/chain-adapters/src/cosmossdk/CosmosSdkBaseAdapter.ts]: yeah good call, make it pretty here: https://github.com/shapeshift/lib/pull/1227/commits/eb271d78e0e6012261611c6e02fb0da8cb1efd7e

**Review by 0xApotheosis — APPROVED** [review]
Conceptually solid, a few comments/questions.

File: packages/chain-adapters/src/cosmossdk/cosmos/CosmosChainAdapter.ts [hunk 2]
```diff
@@ -16,14 +16,16 @@ import {
...
-import { toAddressNList } from '../../utils'
+import { bn, calcFee, toAddressNList } from '../../utils'
 import {
   assertIsValidatorAddress,
   ChainAdapterArgs,
   CosmosSdkBaseAdapter,
 } from '../CosmosSdkBaseAdapter'
 import { Message, ValidatorAction } from '../types'
 
+export const MIN_FEE = '2500'
```
[the remark this answers is not in the document]
Comment (kaladinlight) [reply packages/chain-adapters/src/cosmossdk/cosmos/CosmosChainAdapter.ts:27]: most likely they can diverge, we are just happening to use the same value right now

File: packages/chain-adapters/src/cosmossdk/cosmos/CosmosChainAdapter.ts [hunk 3]
```diff
@@ -275,12 +277,15 @@ export class ChainAdapter extends CosmosSdkBaseAdapter<KnownChainIds.CosmosMainn
...
+
     // We currently don't have a way to query validators to get dynamic fees, so they are hard coded.
     // When we find a strategy to make this more dynamic, we can use 'sendMax' to define max amount.
     return {
-      fast: { txFee: '5000', chainSpecific: { gasLimit: '250000' } },
-      average: { txFee: '3500', chainSpecific: { gasLimit: '250000' } },
-      slow: { txFee: '2500', chainSpecific: { gasLimit: '250000' } },
+      fast: { txFee: calcFee(MIN_FEE, 'fast', scalars), chainSpecific: { gasLimit } },
+      average: { txFee: calcFee(MIN_FEE, 'average', scalars), chainSpecific: { gasLimit } },
+      slow: { txFee: calcFee(MIN_FEE, 'slow', scalars), chainSpecific: { gasLimit } },
```
[the remark this answers is not in the document]
Comment (kaladinlight) [reply packages/chain-adapters/src/cosmossdk/cosmos/CosmosChainAdapter.ts:288]: typed here using string union https://github.com/shapeshift/lib/pull/1227/files#diff-0a5d1a538ef20b20405222656c9630308f0a05c8d0a0413cf9dec3e3a323b41bR6

**Pull request merged**